### ISSUE TITLE
False-negative: HexLiteralCase does not throw violation on Hex float literals

### ISSUE THREAD

**Issue by Aman-Baliyan** [checkstyle/checkstyle#18653]
I have read check documentation: https://checkstyle.org/checks/misc/hexliteralcase.html#HexLiteralCase
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

```bash
/var/tmp $ javac Test.java

PS C:\Users\amanc\Downloads\Ai_Interviewer\PracticeCheckStyle\src\temp> javac Test.java
PS C:\Users\amanc\Downloads\Ai_Interviewer\PracticeCheckStyle\src\temp>

/var/tmp $ cat config.xml

<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
        "https://checkstyle.org/dtds/configuration_1_3.dtd">

<module name="Checker">
    <property name="localeLanguage" value="en"/>

    <module name="TreeWalker">
       <module name="HexLiteralCase">
       </module>
    </module>
</module>


/var/tmp $ cat Test.java

package temp;
public class Test {
    float a = 0x1a.8p1f; // expected violation for lowercase a
    float b = 0x1A.8p1f; // correct
    double c = 0x1c.8p1d; // expected violation for lowercase c
    double d = 0x1C.8p1d; // correct
}


/var/tmp $ RUN_LOCALE="-Duser.language=en -Duser.country=US"
/var/tmp $ java $RUN_LOCALE -jar checkstyle-13.0.0-all.jar -c config.xml Test.java

Starting audit...
Audit done.

```
---

**Describe what you expect in detail.**
 There should be violations in Test.java for HexLiteralCase but there are no violations . The HexLiteralCase module     does not work for hex float literals

---

**Comment by zyadahmed** [comment]
Yes, that’s correct. In the initial design of this check, we didn’t take hexadecimal floating-point literals into account. Do you think we should approve this, @romani ?

**Comment by Aman-Baliyan** [comment]
if this is approved then i am intrested in working on it

**Comment by navin-oss** [comment]
Thanks for the clarification.
I agree that hex floating-point literals were not considered in the initial design.
I’d be interested in contributing a fix if this issue gets approved.

**Comment by romani** [comment]
I was no even aware that it is possible :), but ok , Check should be extended to support `hexadecimal floating-point literals`

please do not forget to update Examples, I doubt a lot of people knows that it is possible to have  `hexadecimal floating-point literals`.

**Comment by Aman-Baliyan** [comment]
ok, i am on it

**Comment by romani** [comment]
Fix is merged.
